### PR TITLE
Fix: correctly handle that lldp_loc_man_addr contains only IPv6 address without IPv4 address

### PR DESCRIPTION
**Pull request by qiluo-msft**
Fixed https://github.com/Azure/sonic-snmpagent/issues/163

### CODE REVIEW

**Comment by lguohan** [comment]
what is the miss here? can you add unit test to prevent future regression?

**Comment by qiluo-msft** [comment]
The issue is that LLDP_LOC_CHASSIS.lldp_loc_man_addr has only one IPv6 address, which happens outside the snmp container. The fix is just to handle this unexpected data peacefully.

We are trying to figure out a way to unit test.

**Comment by lguohan** [comment]
what changed? is this a regression?

**Comment by qiluo-msft** [comment]
I have not investigate outside snmp container. I believe it is an image regression issue.

**Comment by lguohan** [comment]
> The issue is that LLDP_LOC_CHASSIS.lldp_loc_man_addr has only one IPv6 address, which happens outside the snmp container

how many ipv6 addresses does this have before?

**Review by lguohan — APPROVED** [review]

**Comment by qiluo-msft** [comment]
Sorry for misleading. The statement should be:
The issue is that LLDP_LOC_CHASSIS.lldp_loc_man_addr has no IPv4 address(es) at all but only one or more IPv6 address(es), which happens outside the snmp container. The fix is just to handle this unexpected data peacefully.

**Comment by lguohan** [comment]
what is the situation before?

**Comment by qiluo-msft** [comment]
In a good system, I see one IPv4 plus one IPv6 address.

**Comment by lguohan** [comment]
make sense, can you put this into the issue and ask @gen-hwa to look at it further? why we are missing ipv4 address now?

File: src/sonic_ax_impl/mibs/ieee802_1ab.py [hunk 1]
```diff
@@ -334,12 +337,12 @@ def update_data(self):
 
     def get_next(self, sub_id):
         right = bisect_right(self.man_addr_list, sub_id)
-        if right == len(self.man_addr_list):
+        if right >= len(self.man_addr_list):
```
Comment (SuvarnaMeenakshi) [inline src/sonic_ax_impl/mibs/ieee802_1ab.py:340]: Why is this change required?

File: src/sonic_ax_impl/mibs/ieee802_1ab.py [hunk 2]
```diff
@@ -317,6 +317,9 @@ def reinit_data(self):
                 if '.' in mgmt_ip:
                     mgmt_ip_sub_oid = (addr_subtype_sub_oid, *[int(i) for i in mgmt_ip.split('.')])
                     break
+            else:
```
Comment (SuvarnaMeenakshi) [inline src/sonic_ax_impl/mibs/ieee802_1ab.py:320]: else block is not intended to match the right if block.

File: src/sonic_ax_impl/mibs/ieee802_1ab.py (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic_ax_impl/mibs/ieee802_1ab.py:340]: This is not required. It just make code more robust.

---
In reply to: [503552127](https://github.com/Azure/sonic-snmpagent/pull/164#discussion_r503552127) [](ancestors = 503552127)

File: src/sonic_ax_impl/mibs/ieee802_1ab.py (hunk 2 above)
Comment (qiluo-msft) [reply src/sonic_ax_impl/mibs/ieee802_1ab.py:320]: This is for-else structure, which is a critical change in this PR. If there is no IPv4 address matched in the for loop, we need to return soon. Otherwise future statement will fail since mgmt_ip_sub_oid==None

---
In reply to: [503552324](https://github.com/Azure/sonic-snmpagent/pull/164#discussion_r503552324) [](ancestors = 503552324)

Comment (SuvarnaMeenakshi) [reply src/sonic_ax_impl/mibs/ieee802_1ab.py:320]: got it, thank you.

**Review by SuvarnaMeenakshi — APPROVED** [review]
